File: Vireo3/test/org/tdl/vireo/model/MockEmailTemplate.java
package org.tdl.vireo.model;

/**
 * This is a simple mock email template class that may be useful for testing.
 * Feel free to extend this to add in extra parameters that you feel
 * appropriate.
 * 
 * The basic concept is all properties are public so you can create the mock
 * object and then set whatever relevant properties are needed for your
 * particular test.
 * 
 * @author <a href="http://www.scottphillips.com">Scott Phillips</a>
 */
public class MockEmailTemplate extends AbstractMock implements EmailTemplate {

	/* Email Template Properties */
	public int displayOrder;
	public String name;
	public String subject;
	public String message;
	public boolean required;

	@Override
	public MockEmailTemplate save() {
		return this;
	}

	@Override
	public MockEmailTemplate delete() {
		return this;
	}

	@Override
	public MockEmailTemplate refresh() {
		return this;
	}

	@Override
	public MockEmailTemplate merge() {
		return this;
	}
	
	@Override
	public MockEmailTemplate detach() {
		return this;
	}

	@Override
	public int getDisplayOrder() {
		return displayOrder;
	}

	@Override
	public void setDisplayOrder(int displayOrder) {
		this.displayOrder = displayOrder;
	}

	@Override
	public String getName() {
		return name;
	}

	@Override
	public void setName(String name) {
		this.name = name;
	}
	
	@Override
	public String getSubject() {
		return subject;
	}

	@Override
	public void setSubject(String subject) {
		this.subject = subject;
	}

	@Override
	public String getMessage() {
		return message;
	}

	@Override
	public void setMessage(String message) {
		this.message = message;
	}
	
	@Override
	public boolean isSystemRequired() {
		return required;
	}
	
	@Override
	public void setSystemRequired(boolean required) {
		this.required = required;
	}

}

File: src/main/webapp/tests/mocks/models/mockEmailTemplate.js
var dataEmailTemplate1 = {
    id: 1,
    name: "name 1",
    message: "message 1",
    subject: "subject 1",
    systemRequired: true

};

var dataEmailTemplate2 = {
    id: 2,
    name: "name 2",
    message: "message 2",
    subject: "subject 2",
    systemRequired: false
};

var dataEmailTemplate3 = {
    id: 3,
    name: "name 3",
    message: "message 3",
    subject: "subject 1",
    systemRequired: false
};

var dataEmailTemplate4 = {
    id: 4,
    name: "name 4",
    message: "message 2",
    subject: "subject 2",
    systemRequired: true
};

var dataEmailTemplate5 = {
    id: 5,
    name: "name 5",
    message: "message 3",
    subject: "subject 5",
    systemRequired: false
};

var dataEmailTemplate6 = {
    id: 6,
    name: "name 6",
    message: "message 6",
    subject: "subject 6",
    systemRequired: false
};

var mockEmailTemplate = function($q) {
    var model = mockModel("EmailTemplate", $q, dataEmailTemplate1);

    return model;
};

angular.module("mock.emailTemplate", []).service("EmailTemplate", mockEmailTemplate);

